File: P2-Linguagem_de_Programação_II/Roteiros de Aula - C++/Roteiro de Aula 07/cachorro.cpp
#include <iostream>
#include <string>

class Cachorro {
public:
    std::string nome;
    std::string raca;
    int idade;

    Cachorro(std::string n, std::string r, int i) {
        nome = n;
        raca = r;
        idade = i;
    }

    std::string latir() {
        return nome + " está latindo: Au Au!";
    }

    std::string comer(std::string comida) {
        return nome + " está comendo " + comida + ".";
    }
};

int main() {

    Cachorro cachorro1("Rex", "Labrador", 3);
    Cachorro cachorro2("Bolinha", "Poodle", 5);

    std::cout << "Nome: " << cachorro1.nome << ", Raca: " << cachorro1.raca << ", Idade: " << cachorro1.idade << " anos" << std::endl;
    std::cout << "Nome: " << cachorro2.nome << ", Raca: " << cachorro2.raca << ", Idade: " << cachorro2.idade << " anos" << std::endl;

    std::cout << cachorro1.latir() << std::endl;
    std::cout << cachorro2.comer("ração") << std::endl;

    return 0;
}

File: P2-Linguagem_de_Programação_II/Roteiros de Aula - Java/Roteiro de Aula 07/Cachorro.java
public class Cachorro {
    String nome;
    int idade;
    String raca;

    public Cachorro(String nome, String raca, int idade) {
        this.nome = nome;
        this.raca = raca;
        this.idade = idade;
    }

    public String latir() {
        return nome + " está latindo: Au Au!";
    }

    public String comer(String comida) {
        return nome + " está comendo " + comida + ".";
    }

    public static void main(String[] args) {
        Cachorro cachorro1 = new Cachorro("Rex", "Labrador", 3);
        Cachorro cachorro2 = new Cachorro("Bolinha", "Poodle", 5);

        System.out.println("Nome: " + cachorro1.nome + ", Raca: " + cachorro1.raca + ", Idade: " + cachorro1.idade + " anos");
        System.out.println("Nome: " + cachorro2.nome + ", Raca: " + cachorro2.raca + ", Idade: " + cachorro2.idade + " anos");

        System.out.println(cachorro1.latir());
        System.out.println(cachorro2.comer("ração"));
    }
}